pkg/jwt/provider/verifier_test.go

```diff
@@ -0,0 +1,97 @@
+package provider
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	fakeVerifier1 *mockVerifier
+	fakeVerifier2 *mockVerifier
+
+	r              *http.Request
+	verifierBasket Verifier
+)
+
+type mockVerifier struct {
+	result1 bool
+	result2 error
+}
+
+func (v *mockVerifier) Verify(r *http.Request) (bool, error) {
+	return v.result1, v.result2
+}
+
+func (v *mockVerifier) VerifyReturns(result1 bool, result2 error) {
+	v.result1 = result1
+	v.result2 = result2
+}
+
+func TestAllVerifiersFailed(t *testing.T) {
```

Comment:
We could start using table tests, we have a example on push-notification-service